jsk_panda_robot/panda_eus/euslisp/dual_panda-interface.l

```diff
@@ -0,0 +1,380 @@
+           (return)))
+       ) ;; when gazebo
+     (unless gazebop
+       (ros::ros-warn "real franka environemt interface is not impelemented")) ;; TODO
+
+     ;; for error recovery
+     (ros::create-nodehandle "error_group")
+     (ros::subscribe "/dual_panda/rarm/has_error" std_msgs::Bool
+                     #'send self :callback-rarm-error 1 :groupname "error_group")
+     (ros::subscribe "/dual_panda/larm/has_error" std_msgs::Bool
+                     #'send self :callback-larm-error 1 :groupname "error_group")
+     (setq error-recovery-act (instance ros::simple-action-client :init
+                                        "/dual_panda/error_recovery"
+                                        franka_msgs::ErrorRecoveryAction
+                                        :groupname "error_group"
+                                        ))
+     ;; actions for gripper
+     (setq r-gripper-grasp-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/rarm/franka_gripper/grasp"
+                     franka_gripper::GraspAction))
+     (setq r-gripper-homing-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/rarm/franka_gripper/homing"
+                     franka_gripper::HomingAction))
+     (setq r-gripper-move-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/rarm/franka_gripper/move"
+                     franka_gripper::MoveAction))
+     (setq r-gripper-stop-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/rarm/franka_gripper/stop"
+                     franka_gripper::StopAction))
+
+     (setq l-gripper-grasp-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/larm/franka_gripper/grasp"
+                     franka_gripper::GraspAction))
+     (setq l-gripper-homing-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/larm/franka_gripper/homing"
+                     franka_gripper::HomingAction))
+     (setq l-gripper-move-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/larm/franka_gripper/move"
+                     franka_gripper::MoveAction))
+     (setq l-gripper-stop-action
+           (instance ros::simple-action-client :init
+                     "/dual_panda/larm/franka_gripper/stop"
+                     franka_gripper::StopAction))
+     ))
+  (:default-controller
+   ()
+   (list
+    (list
+     (cons :controller-action "/dual_panda/dual_panda_effort_joint_trajectory_controller/follow_joint_trajectory")
+     (cons :controller-state  "/dual_panda/dual_panda_effort_joint_trajectory_controller/state")
+     (cons :action-type control_msgs::FollowJointTrajectoryAction)
+     (cons :joint-names (send-all (send robot :joint-list) :name))
+     )))
+  #|
+  (:default-controller ;; for gazebo
```

Comment:
can we remove this line?

Reply:
Sorry, I didn't check gazebo at all.
I'll check it and check what is happening.

Reply:
Removed because more effort is required for restoring gazebo and we currently don't use gazebo.